pandas/core/internals/managers.py

```diff
@@ -584,6 +584,7 @@ def convert(self, **kwargs):
         return self.apply("convert", **kwargs)
 
     def replace(self, **kwargs):
+        assert "value" in kwargs and np.ndim(kwargs["value"]) == 0
```

Comment:
Does adding this to the signature explicitly not work?

Reply:
I think it does, but the pattern for the other self.aply usages here is just to have kwargs.  I'm OK either way

Reply:
Yea I think signature would be better if required

Reply:
changed